src/AvatarStack.js

```diff
@@ -1,147 +1,149 @@
-      outline: ${get('borderWidths.1')} solid ${get('colors.white')};
-    }
-
-    &::before {
-      width: 17px;
-      background: ${get('colors.gray.2')};
+  &.pc-AvatarStack--two {
+    min-width: 30px;
+    .pc-AvatarItem {
+      &:nth-child(n + 3) {
+        display: none;
+      }
     }
+  }
 
-    &::after {
-      width: 14px;
-      background: ${get('colors.gray.3')};
+  &.pc-AvatarStack--three-plus {
+    min-width: 38px;
+    .pc-AvatarItem {
+      &:nth-child(3) {
+        opacity: ${100 - 3 * 15}%;
+        margin-left: -17px;
+      }
+      &:nth-child(4) {
+        opacity: ${100 - 4 * 15}%;
+        margin-left: -17px;
+      }
+      &:nth-child(5) {
+        opacity: ${100 - 5 * 15}%;
+        margin-left: -17px;
+      }
+      &:nth-child(n + 4) {
+        display: block;
+      }
+      &:nth-child(n + 6) {
+        opacity: 0;
+        visibility: hidden;
+      }
     }
   }
 
-  ${props =>
-    props.alignRight &&
-    css`
-      right: 0;
-      flex-direction: row-reverse;
+  &.pc-AvatarStack--right {
+    justify-content: flex-end;
+    .pc-AvatarItem {
+      margin-left: 0 !important;
+      margin-right: -11px;
 
-      &:hover .AvatarItem {
+      &:first-child {
         margin-right: 0;
-        margin-left: 3px;
       }
+    }
 
-      .AvatarItem-more {
-        background: ${get('colors.gray.3')};
+    .pc-AvatarStackBody {
+      flex-direction: row-reverse;
 
-        &::before {
-          width: 5px;
-        }
+      &:hover {
+        .pc-AvatarItem {
+          margin-right: ${get('space.1')}!important;
+          margin-left: 0 !important;
 
-        &::after {
-          background: ${get('colors.gray.1')};
-          width: 2px;
+          &:first-child {
+            margin-right: 0;
+          }
         }
       }
+    }
+  }
 
-      .AvatarItem {
-        margin-right: 0;
-        margin-left: -11px;
-        border-right: 0;
-        border-left: ${get('borderWidths.1')} solid ${get('colors.white')};
+  &.pc-AvatarStack--three-plus.pc-AvatarStack--right {
+    .pc-AvatarItem {
+      &:nth-child(3) {
+        margin-right: -17px;
+      }
+      &:nth-child(4) {
+        margin-right: -17px;
+      }
+      &:nth-child(5) {
+        margin-right: -17px;
+      }
+    }
+  }
+
+  ${COMMON}
+  ${sx};
+`
+
+const AvatarStackBody = styled.span`
+  display: flex;
+  position: absolute;
+  width: 38px;
+
+  &:hover {
+    width: auto;
+
+    .pc-AvatarItem {
+      margin-left: ${get('space.1')}!important;
+      opacity: 100% !important;
+      visibility: visible;
+      box-shadow: 0 0 0 4px ${get('colors.white')};
+      &:first-child {
+        margin-left: 0;
       }
-    `}
+    }
+  }
 `
 const AvatarStack = ({children = [], alignRight, ...rest}) => {
+  const count = children.length
+  const wrapperClassNames = classnames(
+    count === 2 ? 'pc-AvatarStack--two' : count > 2 ? 'pc-AvatarStack--three-plus' : '',
+    alignRight ? 'pc-AvatarStack--right' : ''
+  )
```

Comment:
This nested ternary is pretty funky; `classnames` supports a syntax that will allow us to unwrap it:

```suggestion
  const wrapperClassNames = classnames({
    'pc-AvatarStack--two': count === 2,
    'pc-AvatarStack--three-plus': count > 2,
    'pc-AvatarStack--right': alignRight
  })
```

Reply:
whaaaa this is awesome!